### PR TITLE
Documentation on how to write Fido clients

### PR DESCRIPTION
**Pull request by Cadair**
TODO:

- [ ] Document how `_can_hancle_query` gets all the `AttrOr` and `AttrAnds` removed before it is called.
- [ ] Document how to add attrs to the registry.

### CODE REVIEW

File: sunpy/net/attr.py [hunk 1]
```diff
@@ -282,7 +282,7 @@ class Range(DataAttr):
     An attribute that represents a range of a value.
 
     This type of attribute would be applicable for types like Wavelength or Time.
-    The range is inclusive of both the min and ma
+    The range is inclusive of both the min and max
```
Comment (nabobalis) [inline sunpy/net/attr.py]:
```suggestion
    The range is inclusive of both the min and max.
```

**Comment by abhijeetmanhas** [comment]
I have added how to write a new "simple" Fido client, w.r.t. #4321 in this gist https://gist.github.com/abhijeetmanhas/92db4933b2ecbd233ca56326c13f7f00. 
Will shortly convert it into a PR.

**Comment by Cadair** [comment]
@abhijeetmanhas Feel free to adopt this branch and make a new PR for it. Would be good to work on both prongs of this documentation at the same time.

**Comment by nabobalis** [comment]
> @abhijeetmanhas Feel free to adopt this branch and make a new PR for it. Would be good to work on both prongs of this documentation at the same time.

You will most likely need to tackle the first bullet point at some point in the future.

**Comment by Cadair** [comment]
Yeah, I am not suggesting @abhijeetmanhas writes all of both sides, just I can push to his PR but he can't push to mine.

**Comment by nabobalis** [comment]
Closed in favour of #4387

**Pull request closed without merging**